### ISSUE TITLE
Password recovery quirks as discovered by Em

### ISSUE THREAD

**Issue by emilyashley** [MapStory/mapstory#895]
So this is kinda funny, but something we could address. 
Here's my most recent interaction with the password reset process.

From the login modal, I type in my username and password. 
It's incorrect. 
It redirects to `account/login/?next=%2Faccount%2Fsignup%2F` 

It says 1 thing: **The username and/or password you specified are not correct.**

I assume this just means I haven't made an account on beta. 
So, using the signup modal, I enter my desired information.
it redirects to `/account/signup/` which says 2 things.

1. **my username is already taken**
2. **A user is registered with this email address**

Ok, so I must've just forgotten my password.
I clicked 'Forgot your password'
 `/account/password/reset/`
I type in my email and click reset my password. 

I receive an email, 
and reset my password. 
It was successfully changed. 

I type in my username and new password

**The username and/or password you specified are not correct.**

I did this about 4 or 5 times. 
Why wasn't it working, you ask?

Well, because I didn't _actually_ know my correct username. 

tldr:
## Problem: 
sometimes users don't remember the username associated with an email address

## Possible Solutions:
(in no particular order of validity)
- we could allow signing in with a username OR email
- we could validate whether or not a username _exists_ (not just whether the password matches)

- we could include the username in the password reset email
- we could include the username on the new password template view
- we could pre-populate the username in the sign in form AFTER this reset password view

- we could offer a 'forgot my username' link that sends an email with the username

**Comment by emilyashley** [comment]
![screen shot 2017-08-03 at 5 16 13 pm](https://user-images.githubusercontent.com/15912063/28949979-3ba4cef2-7875-11e7-9a70-d464cbd25a3b.png)

![screen shot 2017-08-03 at 5 38 01 pm](https://user-images.githubusercontent.com/15912063/28949990-4ac9507e-7875-11e7-8928-a340491b9c5d.png)

![screen shot 2017-08-03 at 5 01 21 pm](https://user-images.githubusercontent.com/15912063/28949995-4fa7808e-7875-11e7-9025-c9ebb0736c0a.png)

![screen shot 2017-08-03 at 5 18 50 pm](https://user-images.githubusercontent.com/15912063/28950000-5853272e-7875-11e7-880d-0e0c17b0d1eb.png)
